src/content-handlers/iiif/modules/uv-shared-module/BaseExtension.ts

```diff
@@ -825,6 +815,39 @@ export class BaseExtension<T extends BaseConfig> implements IExtension {
     };
   }
 
+  getAppUri(): string {
+    const options = this.data.config!.modules.shareDialogue.options;
+
+    const host =
+      options?.embedHost ??
+      `${window.location.protocol}//${window.location.hostname}`;
+    const port = options?.embedPort ?? window.location.port;
```

Comment:
Do we need to add logic to leave off default ports like 80 and 443 when they match the protocol, or does this already happen automatically somehow? (I'm guessing more work may be needed, since the demo has a custom-configured port, so you wouldn't see how this behaves in standard HTTP/S by default).

Reply:
It looks like JS takes care of that for us: https://developer.mozilla.org/en-US/docs/Web/API/Location/protocol.

> If the port is the default for the protocol (80 for ws: and http:, 443 for wss: and https:, and 21 for ftp:), this property contains an empty string, "".

Reply:
Hooray for well-designed magic!